src/main/kotlin/tech/harmonysoft/oss/gradle/dist/BuildCustomGradleDistributionTask.kt

```diff
@@ -126,7 +153,8 @@ abstract class BuildCustomGradleDistributionTask : DefaultTask() {
     }
 
     private fun getDistributions(): Collection<String> {
-        val childDirectories = extensionsRootDir.listFiles(FileFilter { it.isDirectory })
+        val extensionsRootDir = this.extensionsRootDir.orNull?.asFile
```

Comment:
it's worth to fail with corresponding error message if this directory doesn't exist - the plugin is pointless otherwise

Reply:
I forced `extensionsRootDir` to be required and the indicated directory to exist. Gradle would return the correct error message to the user if the directory did not exist. For all test cases to succeed, I had to add two init.d directories with the `.gitkeep` file.

```quote
* What went wrong:
A problem was found with the configuration of task ':buildGradleDist' (type 'BuildCustomGradleDistributionTask').
  - In plugin 'tech.harmonysoft.oss.custom-gradle-dist-plugin' type 'tech.harmonysoft.oss.gradle.dist.BuildCustomGradleDistributionTask' property 'extensionsRootDir' specifies directory '/tmp/junit8917516433528672602/input7686248079884748448/src/main/resources/init.d' which doesn't exist.
    
    Reason: An input file was expected to be present but it doesn't exist.
    
    Possible solutions:
      1. Make sure the directory exists before the task is called.
      2. Make sure that the task which produces the directory is declared as an input.
    
    For more information, please refer to https://docs.gradle.org/8.4/userguide/validation_problems.html#input_file_does_not_exist in the Gradle documentation.
```